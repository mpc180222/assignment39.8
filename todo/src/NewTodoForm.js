import React, { useState } from 'react';
import TodoList from './TodoList';

const NewTodoForm = ({addTodo}) => {

    const INITIAL_STATE = {
        todo: ''
    }

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleSubmit = (e) => {

        e.preventDefault();
        addTodo({todo: formData.todo})
        setFormData(INITIAL_STATE);
    }

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData, [name]: value
        }))

    }

    return(

        <form onSubmit = {handleSubmit}>
            <input data-testid="new-todo-input" name = "todo" placeholder = "new todo" value = {formData.todo} onChange = {handleChange}></input>    
        <button data-testid = "submit-todo-input">Add todo</button>
        </form>


    )


}

export default NewTodoForm;