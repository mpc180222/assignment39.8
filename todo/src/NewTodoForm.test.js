import React from 'react';
import { render, screen } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it('renders without crashing', () => {
  render(<NewTodoForm />);
  
});

it("should match snapshot", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();

})