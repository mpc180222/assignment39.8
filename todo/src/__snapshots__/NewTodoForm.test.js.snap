// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      data-testid="new-todo-input"
      name="todo"
      placeholder="new todo"
      value=""
    />
    <button
      data-testid="submit-todo-input"
    >
      Add todo
    </button>
  </form>
</DocumentFragment>
`;
